In [1]:
!pip install pandas

In [6]:
import pandas as pd

In [7]:
df_diabetes = pd.read_csv("https://raw.githubusercontent.com/gazzola/pima_dataset/main/diabetes.csv")

In [8]:
df_diabetes

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [9]:
df_diabetes.head(10)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   
5            5      116             74              0        0  25.6   
6            3       78             50             32       88  31.0   
7           10      115              0              0        0  35.3   
8            2      197             70             45      543  30.5   
9            8      125             96              0        0   0.0   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1  
5                     0.201   30        0  
6                     0.248   26        1  
7                     0.134   29        0  
8                     0.158   53        1  
9                     0.232   54        1

In [10]:
df_diabetes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [11]:
df_diabetes.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [12]:
df_diabetes.shape

(768, 9)

**O DataFrame tem 768 linhas e 9 colunas.**

In [20]:
df_diabetes = df_diabetes.rename(columns={"Pregnancies": "Gravidez", "Glucose":"Glicose", "BloodPressure":"Pressão Arterial", "SkinThickness":"Espessura da Pele", "Insulin":"Insulina", "BMI":"IMC", "DiabetesPedigreeFunction":"Função Pedigree Diabetes", "Age":"Idade", "Outcome":"Resultado"})

#df = df.rename(columns={'oldName1': 'newName1', 'oldName2': 'newName2'})

In [21]:
df_diabetes.describe()

Gravidez     Glicose  Pressão Arterial  Espessura da Pele  \
count  768.000000  768.000000        768.000000         768.000000   
mean     3.845052  120.894531         69.105469          20.536458   
std      3.369578   31.972618         19.355807          15.952218   
min      0.000000    0.000000          0.000000           0.000000   
25%      1.000000   99.000000         62.000000           0.000000   
50%      3.000000  117.000000         72.000000          23.000000   
75%      6.000000  140.250000         80.000000          32.000000   
max     17.000000  199.000000        122.000000          99.000000   

         Insulina         IMC  Função Pedigree Diabetes       Idade  \
count  768.000000  768.000000                768.000000  768.000000   
mean    79.799479   31.992578                  0.471876   33.240885   
std    115.244002    7.884160                  0.331329   11.760232   
min      0.000000    0.000000                  0.078000   21.000000   
25%      0.000000   27.300000                  0.243750   24.000000   
50%     30.500000   32.000000                  0.372500   29.000000   
75%    127.250000   36.600000                  0.626250   41.000000   
max    846.000000   67.100000                  2.420000   81.000000   

        Resultado  
count  768.000000  
mean     0.348958  
std      0.476951  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      1.000000

In [22]:
analysis = {"Pregnancies":"Analisar", "Glucose":"Analisar", "BP":"Analisar", "ST":"Analisar", "Insulin":"Analisar", "BMI":"Analisar", "DPF":"Analisar", "Age":"Analisar", "Outcome":"Analisar"}

**Pregnancies** variable makes sense: minimum is 0 and
maximum is 17.

**Glucose** variable doesn't make sense: minimum is 0 (not possible) and maximum is 199 (normal, would qualify as pre-diabetes).
- Meaning there is an **error** in the data.

**Blood pressure** variable doesn't make sense: minimum is 0 (not possible) and maximum is 122 (normal).
- Meaning there is an **error** in the data.

**Skin thickness** variable doesn't make sense: minimum is 0 (not possible for humans) and maximum is 99 (normal).
- Meaning there is an **error** in the data.

**Insulin variable** doesn't make complete sense: minimum is 0 (technically possible but unlikely), and maximum is 846 (meaning diabetes, but normal within that range)
- Meaning there is **probably an error** in the data

**BMI** variable doesn't make sense: minimum is 0 (not possible) and maximum is 67 (would indicate obesity, but it's possible)
- Meaning there is an **error** in the data

**Diabetes Pedigree Function** variable makes sense: minimum is 0.07 and maximum is 2.42 (normal being 0 to 2.5)

**Age** variable makes sense: minimum is 21 (as the study only required woman above 21) and maximum is 81.

**Outcome** variable makes sense: as minimum is 0 and maximum is 1, the only two values it can be (has diabetes or not)

In [23]:
analysis = {"Pregnancies":"OK", "Glucose":"OK", "BP":"OK", "ST":"OK", "Insulin":"OK", "BMI":"OK", "DPF":"OK", "Age":"OK", "Outcome":"OK"}